### PR TITLE
feat(worker): dynamic content caching plugin

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: service-worker/worker/src/plugins/dynamic/index.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        this.worker.adapter.newResponse('Not Found', 404, 'Not Found');
+  }
+
+  private get config(): DynamicContentCacheManifest {
+    return this.worker.manifest[this.manifestKey] as DynamicContentCacheManifest;
+  }
+
+  private regexp(str: string): RegExp {
+    if (!this.regexpMap.hasOwnProperty(str)) {
+      this.regexpMap[str] = new RegExp(str).compile();
```
Comment (mgechev) [inline service-worker/worker/src/plugins/dynamic/index.ts]: Is the usage of `compile()` intentional? It seems [deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/compile).

File: service-worker/worker/src/plugins/dynamic/index.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  }
+
+  private fetchCacheOnly(req: Request): FetchInstruction {
+    const instruction: FetchInstruction = () => {
+      return Observable.concat(
+        this.worker.cache.load(this.cacheKey, req),
+        Observable.of(this.errNotFound),
+      );
+    };
+    instruction.desc = {type: 'fetchFastest', req, plugin: this};
```
Comment (mgechev) [inline service-worker/worker/src/plugins/dynamic/index.ts]: Shouldn't it be `fetchCacheOnly` instead of `fetchFastest`?

**Review by mgechev — COMMENTED** [review]
Great way to empower different fetching strategies! I left some small comments.

**Comment by jeffbcross** [comment]
Where are the tests?

**Pull request closed without merging**